src/api/app/controllers/webui/users/notifications_controller.rb

```diff
@@ -2,7 +2,43 @@ class Webui::Users::NotificationsController < Webui::WebuiController
   before_action :require_login
 
   def index
-    notification_type = params[:type]
+    @projects_with_count = Notification.for_subscribed_user(User.session).projects_with_count
+
+    if params[:project]
+      notifications_by_project
+    else
+      notifications_by_type
+    end
+  end
+
+  def update
+    notification = User.session.notifications.find(params[:id])
+    authorize notification, policy_class: NotificationPolicy
+
+    if notification.toggle(:delivered).save
+      flash[:success] = "Successfully marked the notification as #{notification.unread? ? 'unread' : 'read'}"
+    else
+      flash[:error] = "Couldn't mark the notification as #{notification.unread? ? 'read' : 'unread'}"
+    end
+    redirect_back(fallback_location: root_path)
+  end
+
+  private
+
+  def notifications_by_project
+    @notifications = Notification.with_notifiable.unread.includes([:notifiable])
+                                 .for_subscribed_user(User.session)
+                                 .select do |notification|
+                                   notification if notification.notifiable.notifiable_projects.include?(params[:project])
+                                 end
+    return if @notifications.present?
+
+    flash[:error] = 'Filter not valid.'
```

Comment:
I would split the logic regarding fetching the `@notifications` from the logic regarding the `flash[:error]`

Reply:
`notifications_by_project` looks like a _getter_ method and making it side-effectful with the `flash` and the `redirect_to` is a bit misleading to me

Reply:
Logic splitted. We created `before_action` methods. Please have a look again.

Reply:
Okay